sockeye/training.py

```diff
@@ -801,7 +802,10 @@ def _cleanup(self, lr_decay_opt_states_reset: str, process_manager: Optional['De
                 decoded_checkpoint, decoder_metrics = result
                 self.state.metrics[decoded_checkpoint - 1].update(decoder_metrics)
                 self.tflogger.log_metrics(decoder_metrics, decoded_checkpoint)
-            utils.write_metrics_file(self.state.metrics, self.metrics_fname)
+
+
+                utils.write_metrics_file(self.state.metrics, self.metrics_fname)
+                self.state.save(os.path.join(self.training_state_dirname, C.TRAINING_STATE_NAME))
```

Comment:
This makes sure that the state is written to disk after updating `self.state.metrics` so that the pickle and the `metrics` text file are in sync.